Code/Tools/FBuild/FBuildTest/Data/TestBFFParsing/if_directive.bff

```diff
@@ -50,6 +50,53 @@ Print( "Platform = $Platform$" )
     #endif
 #endif
 
+// Boolean algebra 
```

Comment:
Rather than extending this existing test file, it'd be preferable to add a new individual test cases to TestBFFParsing.cpp. 
You can use the helper macros to check that individual snippets parse correctly, or that they fail with an expected error.

For example (these won't be exactly right but are just to give the general idea):

```C++
void TestBFFParsing::IfBooleanOperators() const
{
    // Failure cases
    TEST_PARSE_FAIL( "#if ||",          "<expected error>" );
    TEST_PARSE_FAIL( "#if &&",          "<expected error>" );
    TEST_PARSE_FAIL( "#if X && || Y\n"
                     "#endif",          "<expected error>" );
    TEST_PARSE_FAIL( "#if X && Y Z\n",
                     "#endif",          "<expected error>" );

    // OR
    TEST_PARSE_OK( "#define A\n"
                   "#if A || B\n"
                   "\n" // Should be here
                   "#else\n"
                   "    #error Should not be here\n"
                   "#endif" );
    TEST_PARSE_OK( "#if A || B\n"
                   "    #error\n" // Should not be here
                   "#endif" );
   <other cases, negation etc>

    // AND
    <etc>

    // Check precedence
    TEST_PARSE_OK( "#define A\n"
                   "#define B\n"
                   "#if A || B && C\n"
                   "    // Should be here\n"
                   "#else\n"
                   "    #error Should not be here\n"
                   "#endif" );
```